pkg/manifests/manifests.go

```diff
@@ -3458,6 +3631,58 @@ func (f *Factory) ThanosRulerCustomResource(queryURL string, trustedCA *v1.Confi
 	return t, nil
 }
 
+func (f *Factory) mountThanosRulerAlertmanagerSecrets(t *monv1.ThanosRuler, alertmanagerConfig *v1.Secret) {
+	amAuthSecrets := getAdditionalAlertmanagerSecrets(f.config.GetThanosRulerAlertmanagerConfigs())
+	if len(amAuthSecrets) == 0 {
+		return
+	}
+
+	var volumeMounts []v1.VolumeMount
+	var volumes []v1.Volume
+	for i, secret := range amAuthSecrets {
+		volumeName := fmt.Sprintf("alertmanager-additional-config-secret-%d", i)
+		volumes = append(volumes, v1.Volume{
+			Name: volumeName,
+			VolumeSource: v1.VolumeSource{
+				Secret: &v1.SecretVolumeSource{
+					SecretName: secret,
+				},
+			},
+		})
+
+		volumeMounts = append(volumeMounts, v1.VolumeMount{
+			Name:      volumeName,
+			MountPath: "/etc/prometheus/secrets/" + secret,
+		})
+	}
+
+	t.Spec.Volumes = append(t.Spec.Volumes, volumes...)
+	for i, _ := range t.Spec.Containers {
+		containerName := t.Spec.Containers[i].Name
+		if containerName == "thanos-ruler" {
+			t.Spec.Containers[i].VolumeMounts = append(t.Spec.Containers[i].VolumeMounts, volumeMounts...)
+		}
+	}
+}
+
+func (f *Factory) injectThanosRulerAlertmanagerDigest(t *monv1.ThanosRuler, alertmanagerConfig *v1.Secret) {
+	digest := ""
+	if alertmanagerConfig == nil {
+		return
+	}
+	digestBytes := md5.Sum([]byte(alertmanagerConfig.StringData["alertmanagers.yaml"]))
+	digest = fmt.Sprintf("%x", digestBytes)
+	for i, _ := range t.Spec.Containers {
+		containerName := t.Spec.Containers[i].Name
+		if containerName == "thanos-ruler" {
+			t.Spec.Containers[i].Env = append(t.Spec.Containers[i].Env, v1.EnvVar{
+				Name:  "ALERTMANAGER_CONFIG_SECRET_VERSION",
```

Comment:
Thanos ruler does not refresh its config when the alertmanagers secret changes. Because of this, we need to redeploy the statefulset whenever there is a change in the data of the secret.

Reply:
Added an inline comment :+1: